rfcs/text/0002_encrypted_attributes.md

```diff
@@ -0,0 +1,228 @@
+- Start Date: 2019-03-22
+- RFC PR: https://github.com/elastic/kibana/pull/34526
+- Kibana Issue: (leave this empty)
+
+# Summary
+
+In order to support the action service we need a way to encrypt/decrypt
+attributes on saved objects that works with security and spaces filtering as
+well as performing audit logging. Sufficiently hides the private key used and
+removes encrypted attributes from being exposed through regular means.
+
+# Basic example
+
+Register saved object type with the `encrypted_saved_objects` plugin:
+
+```typescript
+server.plugins.encrypted_saved_objects.registerType({
+  type: 'server-action',
+  attributesToEncrypt: new Set(['credentials', 'apiKey']),
+  // Optional list of attributes that might not contain user specified data.
+  attributesToExcludeFromAAD: new Set(['temporaryState']),
```

Comment:
What's the use case for `attributesToExcludeFromAAD`? And what does `AAD` stand for?

Reply:
It stands for "Additional authenticated data" according to https://github.com/elastic/kibana/pull/33740/files#diff-30038d010fb873cb850c1ac05ce1610aR117

That should be defined somewhere earlier so it doesn't come as a surprise here. I'd still like to understand the use case though.

Reply:
I can remove it from here and leave only in `Detailed design` section if it feels better.